client/landing/stepper/declarative-flow/internals/index.tsx

```diff
@@ -182,46 +150,11 @@ export const FlowRenderer: React.FC< { flow: Flow } > = ( { flow } ) => {
 		}
 	}, [ flow, ref, isFlowStart ] );
 
-	useEffect( () => {
-		// We record the event only when the step is not empty. Additionally, we should not fire this event whenever the intent is changed
-		if ( ! currentStepRoute || ! hasRequestedSelectedSite ) {
-			return;
-		}
-
-		const signupCompleteFlowName = getSignupCompleteFlowNameAndClear();
-		const signupCompleteStepName = getSignupCompleteStepNameAndClear();
-		const isReEnteringStep =
-			signupCompleteFlowName === flow.name && signupCompleteStepName === currentStepRoute;
-		if ( ! isReEnteringStep ) {
-			recordStepStart( flow.name, kebabCase( currentStepRoute ), {
-				intent,
-				is_in_hosting_flow: isAnyHostingFlow( flow.name ),
-				...( design && { assembler_source: getAssemblerSource( design ) } ),
-			} );
-
-			if ( stepOldSlug ) {
-				recordStepStart( flow.name, kebabCase( stepOldSlug ), {
-					intent,
-					is_in_hosting_flow: isAnyHostingFlow( flow.name ),
-					...( design && { assembler_source: getAssemblerSource( design ) } ),
-				} );
-			}
-		}
-
-		// Also record page view for data and analytics
-		const pathname = window.location.pathname || '';
-		const pageTitle = `Setup > ${ flow.name } > ${ currentStepRoute }`;
-		recordPageView( pathname, pageTitle );
-
-		// We leave out intent from the dependency list, due to the ONBOARD_STORE being reset in the exit flow.
-		// This causes the intent to become empty, and thus this event being fired again.
-		// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, [ flow.name, currentStepRoute, hasRequestedSelectedSite ] );
-
 	const assertCondition = flow.useAssertConditions?.( _navigate ) ?? {
 		state: AssertConditionState.SUCCESS,
 	};
 
+	//TODO: Move it to step route
```

Comment:
To be 100% explicit, can we update the comment to mention the specific component?
```suggestion
	// TODO: Move this logic into the StepRoute component.
```

Reply:
Comment remove here [e576a62](https://github.com/Automattic/wp-calypso/pull/91241/commits/e576a62fabfcf7292ce645b26f784a6a4c19c3c9) since I already opened a new PR addressing it. https://github.com/Automattic/wp-calypso/pull/91345